contrib/zenith_test_utils/zenithtest.c

```diff
@@ -117,3 +129,87 @@ clear_buffer_cache(PG_FUNCTION_ARGS)
 
 	PG_RETURN_VOID();
 }
+
+
+/*
+ * A version to read page for at a specific LSN
+ */
+Datum
+get_raw_page_at_lsn(PG_FUNCTION_ARGS)
+{
+	text	   *relname = PG_GETARG_TEXT_PP(0);
+	text	   *forkname = PG_GETARG_TEXT_PP(1);
+	uint32		blkno = PG_GETARG_UINT32(2);
+	uint64 read_lsn = PG_GETARG_INT64(3);
```

Comment:
It would be nice to have an even more low-level function that takes spcnode/dbnod/relfilenode as argument, so that you could request pages for dropped relations too.

Reply:
Would also be nice to have another variant for requesting the latest version. Maybe do that if you pass NULL as read_lsn?